### PR TITLE
Add ONNX parsing for SimplifiedLayerNormalization

### PR DESCRIPTION
**Pull request by turneram**

### CODE REVIEW

**Comment by CharlieL7** [comment]
Where is this spec for this operator?

File: src/onnx/parse_simplified_layer_normalization.cpp [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        if(x_rank < 2 or x_rank > 3)
+        {
+            MIGRAPHX_THROW("PARSE_SIMPLIFIED_LAYER_NORMALIZATION: invalid input shape");
+        }
+
+        auto x_sq = info.add_common_op("mul", x, x);
+        auto rms  = info.add_instruction(make_op("reduce_mean", {{"axes", {axis}}}), x_sq);
+        auto mean = rms;
+        epsilon =
+            (x_dtype == migraphx::shape::half_type and std::abs(epsilon) < 1e-7) ? 1e-7 : epsilon;
```
Comment (CharlieL7) [inline src/onnx/parse_simplified_layer_normalization.cpp:85]: Why are we limiting the epsilon for half type? It looks like a user input

**Review by CharlieL7 — COMMENTED** [review]
Looks fine, minor comments. I would like to see the equation that it's supposed to be doing.

**Comment by turneram** [comment]
> Where is this spec for this operator?

There isn't actually one for SimplifiedLayerNormalization, but [this](https://github.com/microsoft/onnxruntime/blob/main/docs/ContribOperators.md#commicrosoftskipsimplifiedlayernormalization) is the spec for SkipSimplifiedLayerNormalization, which is just add + SLN. That spec does include an optional bias input, but neither of the ORT implementations utilize it, so I omitted it from ours.

**Comment by turneram** [comment]
> Looks fine, minor comments. I would like to see the equation that it's supposed to be doing.

The equation is the same as [RMS LayerNorm](https://arxiv.org/abs/1910.07467).

File: test/onnx/verify/simplified_layer_normalization.cpp [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+                        half{0.3},
+                        half{-0.6},
+                        half{10.0},
+                        half{-1.0},
+                        half{0.0},
+                        half{1.0},
+                        half{1.2},
+                        half{3.2},
+                        half{-4.1},
+                        half{5.3}};
```
Comment (umangyadav) [inline test/onnx/verify/simplified_layer_normalization.cpp:48]: You shouldn't require casting all the elements to half. Same applies to all other places.

File: src/onnx/parse_simplified_layer_normalization.cpp [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+        auto eps    = info.add_literal(migraphx::literal{migraphx::shape{x_dtype}, {epsilon}});
+        rms         = info.add_common_op("add", rms, eps);
+        auto rrms   = info.add_instruction(make_op("rsqrt"), rms);
+        auto result = info.add_common_op("mul", x, rrms);
+        result      = info.add_common_op("mul", result, scale);
+
+        return {result, mean, rrms};
```
Comment (umangyadav) [inline src/onnx/parse_simplified_layer_normalization.cpp:92]: Is this being matched with LayerNorm kernel on the GPU target ?

**Review by TedThemistokleous — APPROVED** [review]

**Review by umangyadav — APPROVED** [review]

**Pull request merged**